examples/binaries/calc-hash/over-blocks/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2022 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+#![no_std]
+#![feature(const_btree_new)]
+
+use codec::{Decode, Encode};
+use shared::{Package, PackageId, PackageWithId};
+
+#[cfg(feature = "std")]
+mod code {
+    include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
+}
+
+#[cfg(feature = "std")]
+pub use code::WASM_BINARY_OPT as WASM_BINARY;
+
+#[cfg(not(feature = "std"))]
+mod wasm {
+    include! {"./code.rs"}
+}
+
+/// Program methods.
+#[derive(Debug, Encode, Decode)]
+pub enum Method {
+    Start(PackageWithId),
+    Refuel(PackageId),
+    Calculate(Package),
```

Comment:
we don't need to waste out computationals for encoding and decoding package structure - they are all in memory always, we just need here package id